uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/task_test.js

```diff
@@ -0,0 +1,56 @@
+define([
+  'common/angularMocks',
+  'tasks-assignments/app'
+], function () {
+  'use strict';
+
+  describe('TaskCtrl', function () {
+    var $scope, $httpBackend;
+
+    beforeEach(module('civitasks.appDashboard'));
+    beforeEach(inject(function ($controller, $rootScope, _$httpBackend_) {
+      $scope = $rootScope.$new();
+      $httpBackend = _$httpBackend_;
+
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+
+      setup();
+
+      $controller('TaskCtrl', {
+        $scope: $scope
+      });
+    }));
+
+    describe("Task date", function () {
+      beforeEach(function () {
+        spyOn($scope, 'updateTask');
+        $scope.$digest();
+      });
+
+      it("Update task date", function () {
+        $scope.task.activity_date_time = new Date();
+        $scope.$digest();
```

Comment:
It needs to be called twice because $scope.$watch triggers when the controller initializes, if I don't call $scope.$digets the cycle wouldn't not initialize properly.